spec/requests/flows/assertion_spec.rb

```diff
@@ -1,33 +1,78 @@
 require 'spec_helper_integration'
 
-feature 'Resource Owner Assertion Flow inproperly set up' do
-  background do
+describe 'Resource Owner Assertion Flow inproperly set up', type: :request do
+  before do
+    config_is_set(:resource_owner_from_assertion) { nil }
     client_exists
     create_resource_owner
   end
 
   context 'with valid user assertion' do
-    scenario "should not issue new token" do
+    it "should not issue new token" do
       expect {
         post assertion_endpoint_url(client: @client, resource_owner: @resource_owner)
       }.to_not change { Doorkeeper::AccessToken.count }
 
-      should_have_json 'error', 'invalid_resource_owner'
-      should_have_json 'error_description', translated_error_message(:invalid_resource_owner)
+      should_have_json 'error', 'invalid_grant'
+      should_have_json 'error_description', translated_error_message(:invalid_grant)
       expect(response.status).to eq(401)
     end
   end
 end
 
-feature 'Resource Owner Assertion Flow' do
-  background do
+describe 'Resource Owner Assertion Flow', type: :request do
+  before do
     config_is_set(:resource_owner_from_assertion) { User.where(assertion: params[:assertion]).first }
     client_exists
     create_resource_owner
   end
 
+  context "with invalid client/application information" do
+
+    it "should not create an access token" do
+      expect {
+        post assertion_endpoint_url(
+          client_id: 'not-real',
+          client_secret: 'not-real',
+          redirect_uri: 'http://fake-redirect.com'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.